packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -2024,13 +2021,7 @@ class _RenderFocusTrap extends RenderProxyBoxWithHitTestBehavior {
   set focusScopeNode(FocusScopeNode value) {
     if (focusScopeNode == value)
       return;
-    focusScopeNode.removeListener(_currentFocusListener);
     _focusScopeNode = value;
-    focusScopeNode.addListener(_currentFocusListener);
-  }
-
-  void _currentFocusListener() {
```

Comment:
For some reason this wasn't giving up to date results when using tab navigation

Reply:
I am not so certain about why we need to remove this code. If this is indeed a bug fixes, we should have a test for this.